lightpath/path.py

```diff
@@ -489,10 +513,15 @@ def _device_moved(self, *args, obj=None, **kwargs):
         else:
             block = math.inf
         # If device is upstream of impediment
-        if obj and obj.md.z <= block:
+        if obj and obj.parent.md.z <= block:
```

Comment:
```suggestion
        if obj is not None and obj.parent.md.z <= block:
```
Side note: the only object which has metadata is the root `Device`, so this is implicitly a component on the device with a lightpath mixin. That's an OK assumption to make here.